toqito/channel_metrics/fidelity_of_separability.py

```diff
@@ -0,0 +1,198 @@
+        from toqito.states import basis
+
+        state = tensor(basis(2, 0), basis(2, 0), basis(2, 0))
+        rho = state*state.conj().T
+        expected_value = fidelity_of_separability(rho, [2,2,2])
+
+    >>> expected_value = 0.9999999979949119
+
+
+    References
+    ==========
+    .. [Hay12] Hayden, Patrick et.al.
+        "Two-message quantum interactive proofs and the quantum separability problem."
+        Proceedings of the 28th IEEE Conference on Computational Complexity, pages 156-167.
+        https://arxiv.org/abs/1211.6120
+
+    .. [Per96] Peres, Asher.
+        "Separability Criterion for Density Matrices"
+        https://arxiv.org/abs/quant-ph/9604005
+
+    .. [Phil23] Philip, Aby et.al.
+        "Quantum Steering Algorithm for Estimating Fidelity of Separability"
+        https://arxiv.org/abs/2303.07911
+
+    .. [TBWat18] Watrous, John.
+        “The Theory of Quantum Information”
+        Cambridge University Press, 2018
+
+    Args:
+        psi: the density matrix for the tripartite state of interest psi_{BAR}
+        psi_dims: the dimensions of System A, B & R in
+            the input state density matrix. It is assumed that the first
+            quantity in this list is the dimension of System B.
+        k: value for k-extendibility
+    Raises:
+        AssertionError:
+            * If the provided dimensions are not for a tripartite density \n
+            matrix
+        TypeError:
+            * If the matrix is not a density matrix (square matrix that is \n
+            PSD with trace 1).
+        TypeError:
+            * If the input state is a mixed state.
+    Returns:
+        Optimized value of the SDP when maximized over a set of linear
+        operators subject to some constraints.
+    """
+    if not is_density(psi):
+        raise ValueError("Provided input state is not a density matrix.")
+    if not len(psi_dims) == 3:
+        raise AssertionError("For Channel SDP: require tripartite state dims.")
+    if not is_pure(psi):
+        raise ValueError("This function only works for pure states.")
+
+    # We first permure psi_{BAR} to psi_{RAB} to simplify the code.
+    psi = permute_systems(psi, [3, 2, 1], psi_dims)
+    dim_b, dim_a, dim_r = psi_dims
+    psi_dims = [dim_r, dim_a, dim_b]
+
+    # List of dimensions [R, A, B, A'] needed to define optimization function.
+    dim_list = [dim_r, dim_a, dim_b, dim_a]
+
+    # Dimension of the Choi matrix of the extended channel.
+    choi_dims = [dim_r] + [dim_a] * k
+
+    # List of extenstion systems and dimension of the Choi matrix.
+    sys_ext = list(range(2, 2 + k - 1))
+    dim_choi = dim_r * (dim_a**k)
+
+    # Projection onto symmetric subspace on AA'.
+    pi_sym = symmetric_projection(dim_a, 2)
+
+    choi = picos.HermitianVariable("S", (dim_choi, dim_choi))
+    choi_partial = picos.partial_trace(choi, sys_ext, choi_dims)
+    sym_choi = symmetric_projection(dim_a, k)
+    problem = picos.Problem(verbosity=2)
```

Comment:
We may wish to have the `verbosity` argument as an optional argument to the function with a default value of `2`.